### PR TITLE
Performance Validators

### PR DESCRIPTION
**Pull request by rbclark**
This PR merges in multiple different sets of work:

1. Performance validators are added (they are run on every PR now)
1. Travis now runs each set of jobs in a separate matrix.
1. Tests are now moved over to factories.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-38
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by rbclark** [comment]
This should be ready for review and merge. The biggest issue we are currently going to have is I worry that https://github.com/projectcypress/cypress/pull/963 is going to probably conflict since this has gem upgrades in it as well. Also the ruby-prof gem seems to be very sensitive to version upgrades of the other testing gems.

File: features/step_definitions/product.rb [hunk 1]
```diff
@@ -138,38 +139,38 @@ def build_product
...
-  page.all('#measure_tabs .ui-tabs-nav a')[3].click
+  page.all('#measure_tabs .ui-tabs-nav a')[1].click
   page.all('input.measure-checkbox')[0].click
   page.click_button 'Add Product'
 end
 
 When(/^the user creates a product with selecting a group of measures$/) do
   steps %( When the user fills out all product information but measures )
-  page.find("[href='#Miscellaneous_div']").click
+  # page.find("[href='#Miscellaneous_div']").click
```
Comment (okeefm) [inline features/step_definitions/product.rb]: If these lines aren't necessary, should they be removed rather than commented out?

File: test/controllers/checklist_tests_controller_test.rb [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
+    vendor_user = FactoryGirl.create(:vendor_user)
+    FactoryGirl.create(:other_user)
+    @user = FactoryGirl.create(:atl_user)
+    product = FactoryGirl.create(:product_static_bundle)
+    @vendor = product.vendor
+    add_user_to_vendor(vendor_user, @vendor)
+    @bundle = product.bundle
+    @test = product.product_tests.create!({ name: 'c1 visual', measure_ids: ['BE65090C-EB1F-11E7-8C3F-9A214CF093AE'] }, ChecklistTest)
+
+    # collection_fixtures('vendors', 'products', 'product_tests', 'users', 'roles', 'bundles', 'measures')
```
Comment (okeefm) [inline test/controllers/checklist_tests_controller_test.rb]: If these lines aren't necessary, should they be removed rather than commented out?

File: test/controllers/products_controller_test.rb [hunk 3]
```diff
@@ -255,14 +262,14 @@ def create_product_with_checklist_test(measure_ids)
     end
   end
 
-  # # # # # # #
-  #   A P I   #
-  # # # # # # #
+  # # # # # # # #
+  # #   A P I   #
```
Comment (okeefm) [inline test/controllers/products_controller_test.rb]: Why did this line and 269 get double-commented?

File: test/controllers/product_tests_controller_test.rb [hunk 4]
```diff
@@ -49,9 +42,9 @@ class ProductTestsControllerTest < ActionController::TestCase
   end
   # need negative tests for user that does not have owner or vendor access
 
-  # # # # # # #
-  #   A P I   #
-  # # # # # # #
+  # # # # # # # #
+  # #   A P I   #
```
Comment (okeefm) [inline test/controllers/product_tests_controller_test.rb]: Looks like this comment block got double-commented?

File: test/controllers/test_executions_controller_test.rb [hunk 5]
```diff
@@ -47,13 +57,13 @@ def setup_c4
...
 
     # add provider performance to each record associated with the measure test
-    provider = Provider.find('53b2c4414d4d32139c730000')
-    test.provider = provider
-    test.save!
-    test.records.each do |record|
-      pp = record.provider_performances.build(provider: provider)
-      pp.save!
-    end
+    # provider = Provider.find('53b2c4414d4d32139c730000')
```
Comment (okeefm) [inline test/controllers/test_executions_controller_test.rb]: If these lines aren't necessary, should they be removed rather than commented out?

File: test/test_helper.rb [hunk 6]
```diff
@@ -1,10 +1,10 @@
 require 'simplecov'
 
-SimpleCov.start 'rails'
+# SimpleCov.start 'rails'
 # MiniTest changes mean that our coverage suddenly dropped, since more controllers are being tested.
 # We dropped the value to 77 to be able to get pull requests pulled in. Needs to be brought back up as coverage goes back up.
-SimpleCov.minimum_coverage 77
-Mongo::Logger.logger.level = Logger::WARN
+# SimpleCov.minimum_coverage 77
```
Comment (okeefm) [inline test/test_helper.rb]: Did we get rid of Simplecov? If so, we should figure out another way to do code coverage, because we probably don't want to lose that feature

Comment (rbclark) [reply test/test_helper.rb]: I'm not actually sure when this happened, I would really suggest we consider https://codecov.io/ since it is free for open source and provides us verification that coverage has not dropped on tests.

Comment (rbclark) [reply test/test_helper.rb]: Also since the tests are run on multiple different workers now it is not possible to merge results natively, we would need codecov for that.
